packages/utilities/src/EnvFileUtils.ts

```diff
@@ -0,0 +1,59 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { closeSync, existsSync, openSync, readFileSync } from "fs";
+import { homedir } from "os";
+import { join } from "path";
+import { ImperativeError } from "../../error";
+
+/**
+ * Utility to load environment JSON files and set variables
+ * @export
+ * @class EnvFileUtils
+ */
+export class EnvFileUtils {
+
+    /**
+     * Check and read in an environment file from the user home directory using the app name
+     * If the file is valid, set the environment variables
+     * If the file is not valid, display an error and continue
+     * @param {string} appName - The application name
+     * @returns {void}
+     * @throws {ImperativeError}
+     */
+    public static setEnvironmentForApp(appName: string) {
+        const expectedFileLocation = this.getEnvironmentFilePath(appName);
+        if (existsSync(expectedFileLocation)) {
```

Comment:
We might be able to skip the `existsSync` check and then call `openSync` directly - I think CodeQL is mostly complaining about the time between the `existsSync` and `openSync` call.

Since we are already catching exceptions from `openSync` and we are only opening the file w/ read access, we should be able to avoid calling `existsSync` here - just an idea, might clear up the CodeQL warning.

Reply:
The goal here was, if there is no file, then the SDK method shouldn't throw an error or attempt to do anything to the user environment because there is nothing to do. If we want the same behavior and don't want to do existsSync, then we will need to specifically check to see if the error is due to a file not being found, and ignore the error if it is that type of error, and surfacing every other type of error.

As far as I am aware, there is no error type that is able to be detected, so we would be required to check that the error code string is ENOENT, which could possibly change in the future, though that is rather unlikely. I am just a bit hesitant to be doing error detection using string comparisons.